### PR TITLE
[e2e] Fix e2e tests on windows

### PR DESCRIPTION
**Pull request by limdauto**
## Description

Currently master is never green. This fixes the issue :) 

Also added a fix to accommodate recent pylint release

## Checklist

- [ ] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro/io/partitioned_data_set.py [hunk 1]
```diff
@@ -439,7 +439,7 @@ def _parse_checkpoint_config(
         if self._credentials:
             default_config[CREDENTIALS_KEY] = deepcopy(self._credentials)
 
-        if CREDENTIALS_KEY in default_config.keys() & checkpoint_config.keys():
+        if CREDENTIALS_KEY in (default_config.keys() & checkpoint_config.keys()):
```
Comment (arnaldog12) [inline kedro/io/partitioned_data_set.py:442]: I may be wrong, but I believe this condition will never be true. I did some tests and looks like the `&` operator returns an empty set. Please, see the screenshot below:

Screen Shot 2021-12-03 at 15 16 41

Comment (limdauto) [reply kedro/io/partitioned_data_set.py:442]: @arnaldog12 Hey these are set-like views, so `&` only returns non empty set if the intersection is not empty.

Screenshot 2021-12-03 at 18 21 14

Comment (limdauto) [reply kedro/io/partitioned_data_set.py]: I think there is a test that covers this case. In this PR I just want to make it compatible with the pylint release 40 minutes ago but I give up. I will file a ticket on pylint.

File: features/environment.py [hunk 2]
```diff
@@ -110,10 +110,10 @@ def _setup_minimal_env(context):
             "pip>=20.0",
             "setuptools>=38.0",
             "wheel",
-            ".",
         ],
         env=context.env,
     )
+    call([context.python, "-m", "pip", "install", "."], env=context.env)
```
Comment (lorenabalan) [inline features/environment.py:116]: Why does this work?... 😂

**Review by lorenabalan — APPROVED** [review]
Thank you!! 🎊

**Review by MerelTheisenQB — APPROVED** [review]
Wow that's the simplest and smartest fix 😂 🥇

**Pull request merged**